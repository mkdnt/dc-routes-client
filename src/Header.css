.header {
  background-color: #ffa69e;
  font-size: larger;
  text-align: left;
  padding: 0 0 0 5px;
  border: 2px solid black;
  margin: 5px;
}
