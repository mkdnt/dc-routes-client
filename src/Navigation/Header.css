.header {
  background-color: #fff1e6;
  font-size: larger;
  text-align: center;
  border: 2px solid #011328;
  color: #b40421;
}
