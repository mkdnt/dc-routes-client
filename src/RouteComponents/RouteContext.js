import React from 'react';

export default React.createContext({
    routes: [],
    addRoute: () => {},
    editRoute: () => {},
    deleteRoute: () => {},
    editing: null
})