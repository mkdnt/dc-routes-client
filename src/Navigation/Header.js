import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './Header.css'

export class Header extends Component {
    render() {
        return (
            <div className='header'>
                <Link to='/' style={{textDecoration: 'none', color: 'inherit'}}>
                    <h1>dc_routes</h1>
                </Link>
            </div>
        )
    }
}

export default Header
