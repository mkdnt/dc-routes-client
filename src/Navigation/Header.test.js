import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom';
import RouteContext from '../RouteComponents/RouteContext';
import Header from './Header';

describe('Header Component', () => {
  describe('Snapshot', () => {
    it('renders the UI as expected', () => {
      const update = renderer.create(<BrowserRouter><Header /></BrowserRouter>)
        expect(update).toMatchSnapshot()
    })
  })
  
  describe('Smoke test', () => {
    it('Renders without crashing', () => {
      const div = document.createElement('div');
      
      const value = {
        routes: [],
        addRoute: () => {},
        editRoute: () => {},
        deleteRoute: () => {}
        }

      ReactDOM.render(
        <BrowserRouter>
          <RouteContext.Provider value={value}>
            <Header />
        </RouteContext.Provider>
        </BrowserRouter>,
      div);
      ReactDOM.unmountComponentAtNode(div);
    })
  })

});