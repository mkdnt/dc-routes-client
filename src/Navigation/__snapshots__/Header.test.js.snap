// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot renders the UI as expected 1`] = `
<div
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "inherit",
        "textDecoration": "none",
      }
    }
  >
    <h1>
      dc_routes
    </h1>
  </a>
</div>
`;
